src/Groups/cosets.jl

```diff
@@ -1,20 +1,35 @@
 # T=type of the group, S=type of the element
 @doc raw"""
-    GroupCoset{T<: Group, S <: GAPGroupElem}
+    GroupCoset{TG <: GAPGroup, TH <: GAPGroup, S <: GAPGroupElem}
 
-Type of group cosets.
-Two cosets are equal if, and only if, they are both left (resp. right)
+Type of right and left cosets of subgroups in groups.
+
+For an element $g$ in a group $G$, and a subgroup $H$ of $G$,
+the set $Hg = \{ hg; h \in H \}$ is a right coset of $H$ in $G$,
+and the set $gH = \{ gh; h \in H \}$ is a left coset of $H$ in $G$.
+
+- [`group(C::GroupCoset)`](@ref) returns $G$.
+
+- [`acting_group(C::GroupCoset)`](@ref) returns $H$.
+
+- [`representative(C::GroupCoset)`](@ref) returns an element
+  (the same element for each call) of `C`.
+
+- [`is_right(C::GroupCoset)`](@ref) and [`is_left(C::GroupCoset)`](@ref)
+  return whether `C` is a right or left coset, respectively.
+
+Two cosets are equal if and only if they are both left or right, respectively,
 and they contain the same elements.
 """
-struct GroupCoset{T<: GAPGroup, S <: GAPGroupElem} 
-   G::T                    # big group containing the subgroup and the element
-   H::GAPGroup             # subgroup (may have a different type)
+struct GroupCoset{TG <: GAPGroup, TH <: GAPGroup, S <: GAPGroupElem}
+   G::TG                   # big group containing the subgroup and the element
+   H::TH                   # subgroup (may have a different type)
    repr::S                 # element
    side::Symbol            # says if the coset is left or right
    X::Ref{GapObj}          # GapObj(H*repr)
 
-   function GroupCoset(G::T, H::GAPGroup, representative::S, side::Symbol) where {T<: GAPGroup, S<:GAPGroupElem}
-     return new{T, S}(G, H, representative, side, Ref{GapObj}())
+   function GroupCoset(G::TG, H::TH, representative::S, side::Symbol) where {TG <: GAPGroup, TH <: GAPGroup, S <:GAPGroupElem}
+     return new{TG, TH, S}(G, H, representative, side, Ref{GapObj}())
```

Comment:
Maybe enforce the type relation(s) here, e.g. something like `@assert S == elem_type(TG)` or so (assuming that's correct)